.github/workflows/ci.yml

```diff
@@ -24,20 +24,37 @@ jobs:
         uses: actions/setup-python@v4
         with:
           python-version: ${{ matrix.python-version }}
-      - name: Install python dependencies
+      - name: Install dependencies
         run: |
           set -xe
           sudo apt-get update
-          sudo apt-get install -y --no-install-recommends libxslt-dev libxml2-dev libxml2-utils zstd
+          sudo apt-get install -y --no-install-recommends libxslt-dev libxml2-dev libxml2-utils meson zstd
+
+          # Patch Meson to fix pypy-3.9 detection.
```

Comment:
Link to a PR or a bug, or something we could follow?

Reply:
It's already fixed upstream, this is just backporting the fix into Ubuntu's old Meson. I'll clarify though.

Reply:
You may also try installing a newer meson via `pip`.

Reply:
Yeah, fetching the patch is a fair suggestion, so I've done that now. I did consider getting a newer Meson with pip, but testing with an older Meson isn't such a bad idea.

Reply:
Testing against older meson was my original assumption for why you wanted to patch it. Conditionally using a newer version of meson only for pypy would be pretty awkward to do with a matrix job.